docs/resources/credential.md

```diff
@@ -0,0 +1,96 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_credential Resource
+
+-> This resource can only be used with a workspace-level provider.
+
+-> This feature is in [Public Preview](https://docs.databricks.com/release-notes/release-types.html).
+
+A credential represents an authentication and authorization mechanism for accessing services on your cloud tenant. Each credential is subject to Unity Catalog access-control policies that control which users and groups can access the credential.
+
+To create credentials, you must be a Databricks account admin or have the `CREATE SERVICE CREDENTIAL` privilege. The user who creates the credential can delegate ownership to another user or group to manage permissions on it
+
+On AWS, the IAM role for a credential requires a trust policy. See [documentation](https://docs.databricks.com/en/connect/unity-catalog/cloud-services/service-credentials.html#step-1-create-an-iam-role) for more details. The data source [databricks_aws_unity_catalog_assume_role_policy](../data-sources/aws_unity_catalog_assume_role_policy.md) can be used to create the necessary AWS Unity Catalog assume role policy.
+
+## Example Usage
+
+For AWS
+
+```hcl
+resource "databricks_credential" "external" {
+  name = aws_iam_role.external_data_access.name
+  aws_iam_role {
+    role_arn = aws_iam_role.external_data_access.arn
+  }
+  purpose = "SERVICE"
+  comment = "Managed by TF"
+}
+
+resource "databricks_grants" "external_creds" {
+  credential = databricks_credential.external.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["ACCESS"]
+  }
+}
+```
+
+For Azure
+
+```hcl
+resource "databricks_credential" "external_mi" {
+  name = "mi_credential"
+  azure_managed_identity {
+    access_connector_id = azurerm_databricks_access_connector.example.id
+  }
+  purpose = "SERVICE"
+  comment = "Managed identity credential managed by TF"
+}
+
+resource "databricks_grants" "external_creds" {
+  credential = databricks_credential.external.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["ACCESS"]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+- `name` - Name of Credentials, which must be unique within the [databricks_metastore](metastore.md). Change forces creation of a new resource.
+- `purpose` - Indicates the purpose of the credential. Can be `SERVICE`.
+- `owner` - (Optional) Username/groupname/sp application_id of the credential owner.
```

Comment:
Afaik, the `owner` can't be set. It's derived from the identity of the user creating the credential

Reply:
unless this also applies to updates

Reply:
this applies to update - and in case of create, we execute an update as well